sdk/python/v1beta1/kubeflow/katib/api/katib_client.py

```diff
@@ -430,7 +701,56 @@ def get_optimal_hyperparameters(self, name=None, namespace=None):
 
         katibexp = self.get_experiment(name, namespace=namespace)
         result = {}
-        result["currentOptimalTrial"] = katibexp.get(
-            "status", {}).get("currentOptimalTrial")
+        result["currentOptimalTrial"] = katibexp.get("status", {}).get(
+            "currentOptimalTrial"
+        )
 
         return result
+
+
+def double(min: float, max: float, step: float = None):
+    """Sample a float value uniformly between `min` and `max`.
+
+    Args:
+        min: Lower boundary for the float value.
+        max: Upper boundary for the float value.
+        step: Step between float values.
+    """
+
+    parameter = models.V1beta1ParameterSpec(
+        parameter_type="double",
+        feasible_space=models.V1beta1FeasibleSpace(min=str(min), max=str(max)),
+    )
+    if step is not None:
+        parameter.feasible_space.step = str(step)
+    return parameter
+
+
+def int(min: int, max: int, step: int = None):
+    """Sample an integer value uniformly between `min` and `max`.
+
+    Args:
+        min: Lower boundary for the integer value.
+        max: Upper boundary for the integer value.
+        step: Step between integer values.
+    """
+
+    parameter = models.V1beta1ParameterSpec(
+        parameter_type="int",
+        feasible_space=models.V1beta1FeasibleSpace(min=str(min), max=str(max)),
+    )
+    if step is not None:
+        parameter.feasible_space.step = str(step)
+    return parameter
+
+
+def categorical(list: List):
```

Comment:
do we need to check if list is `[]` or not ?

Reply:
Should we introduce this check in [Validation Webhook](https://github.com/kubeflow/katib/blob/master/pkg/webhook/v1beta1/experiment/validator/validator.go#L244-L246), so we can have single place for spec validation ?

Reply:
Of course but I'm not sure if we need some fast-validation for the user who are not familiar with katib. Also representing the validation rule in python's stdout would be better for them to find out a mistake.

Reply:
I think if Validation Webhook fails, user also will see an error in the Python's stdout before creating Katib Experiment.
What do you think @anencore94 ?

Reply:
If the validadation webhook's failure message is detailed enough, your suggestion would be much better than implementing the same validation check in both!